python/pyspark/sql/pandas/serializers.py

```diff
@@ -153,14 +157,15 @@ def _create_batch(self, series):
         from pyspark.sql.pandas.types import _check_series_convert_timestamps_internal, \
             _convert_dict_to_map_items
         from pandas.api.types import is_categorical_dtype
-        # Make input conform to [(series1, type1), (series2, type2), ...]
-        if not isinstance(series, (list, tuple)) or \
-                (len(series) == 2 and isinstance(series[1], pa.DataType)):
-            series = [series]
-        series = ((s, None) if not isinstance(s, (list, tuple)) else s for s in series)
 
-        def create_array(s, t):
-            mask = s.isnull()
+        def create_array(s, t: pa.DataType, dt: Optional[DataType] = None):
```

Comment:
It looks like we don't use type hints for internal funcs.

Reply:
Fixed, see https://github.com/apache/spark/pull/32026/files